### PR TITLE
Added support for giving default options for SparkPost API

### PR DESCRIPTION
**Pull request by bbredewold**

### CODE REVIEW

File: src/Transport/SparkPostTransportTrait.php [hunk 1]
```diff
@@ -26,11 +26,13 @@
      *
      * @param  ClientInterface $client
      * @param  string $key
+     * @param  array $options
      */
-    public function __construct(ClientInterface $client, $key)
+    public function __construct(ClientInterface $client, $key, $options)
```
Comment (Lidbetter) [inline src/Transport/SparkPostTransportTrait.php]: Can you change this line to:
`public function __construct(ClientInterface $client, $key, $options = [])`

File: src/SparkpostServiceProvider.php [hunk 2]
```diff
@@ -40,19 +40,20 @@ public function register()
 
             $manager->extend('sparkpost', function() {
 
-                $config = $this->app['config']->get('services.sparkpost', []);
-                $options = isset($config['guzzle']) ? $config['guzzle'] : [];
-                $client = new Client($options);
+                $config             = $this->app['config']->get('services.sparkpost', []);
+                $sparkpost_options  = isset($config['options']) ? $config['options'] : false;
```
Comment (Lidbetter) [inline src/SparkpostServiceProvider.php]: Please can you change default to `[]` (sparkpost options)
and switch to camelCase variable names: `$sparkpostOptions,`$guzzleOptions`

**Comment by Lidbetter** [comment]
I changed a few things and merged, thanks for contributing.

**Comment by bbredewold** [comment]
Great! Thanks for the changes you made.

**Pull request merged**